### PR TITLE
Add examples of better Bash function libraries

### PR DESCRIPTION
**Pull request by mbbx6spp**
Included in this commit are libraries for git, bash and nexus uploading helper functions. These functions are self-contained libraries and can be sourced from other scripts for reasonable level of reuse.

* [x] add inline comments to explain what parts of the code make it more robust or a debugging aid.
* [X] add example usage script

### CODE REVIEW

File: bash/bash.functions [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  local -r path="${1}"
+
+  echo ">> Testing ${path} is executable"
+  test -x "${path}"
+  echo ">> Testing ${path} has valid Bash syntax"
+  bash -n "${path}"
+}
+
+function z_bash_check_scripts() {
+  local -ra scripts=($(cat "${j_ci_dir}/bash.scripts"))
```
Comment (polynomial) [inline bash/bash.functions]: Is the hope that the README would have examples of how to use these? If so is the hope that someone could run through it as a tutorial? Seems cool, if so the $j_ci_dir needs to be set somewhere? or just use pwd?

Comment (mbbx6spp) [inline bash/bash.functions]: Ooops. That leaked through the refactoring process from embedded inside the executable script out into the library. I'll fix that. And yes, I will provide an example script that sources these and then uses the functions appropriate.

**Comment by mbbx6spp** [comment]
Note: I updated the comment to read better that we talked about over IM yesterday. I will also add inline comments to explain why the style/form/structure is useful.

**Comment by mbbx6spp** [comment]
@polynomial I am going to merge the changes in now that I have added the inline comments just to get them out there and continue working on the documentation templates and workflow stuff which I think might help your HBase endeavors. Hopefully. Or be an utter disappointment. :)

**Pull request merged**